packages/core-mobile/app/new/features/buyOnramp/hooks/useServiceProviders.ts

```diff
@@ -0,0 +1,89 @@
+import { useSelector } from 'react-redux'
+import { selectSelectedCurrency } from 'store/settings/currency'
+import { QueryObserverResult, RefetchOptions } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { useOnRampSourceAmount, useOnRampToken } from '../store'
+import {
+  CreateCryptoQuote,
+  CreateCryptoQuoteError,
+  CreateCryptoQuoteErrorCode,
+  CreateCryptoQuoteNotFoundError,
+  Quote
+} from '../types'
+import { useCreateCryptoQuote } from './useCreateCryptoQuote'
+import { useLocale } from './useLocale'
+
+export const useServiceProviders = (
+  shouldCreateCryptoQuote = true
+): {
+  crytoQuotes: Quote[]
+  isLoadingCryptoQuotes: boolean
+  cryptoQuotesError?: {
+    statusCode: CreateCryptoQuoteErrorCode
+    message: string
+  }
+  refetch: (
+    options?: RefetchOptions
+  ) => Promise<QueryObserverResult<CreateCryptoQuote | undefined, Error>>
+  isRefetchingCryptoQuotes: boolean
+} => {
+  const [onRampToken] = useOnRampToken()
+  const selectedCurrency = useSelector(selectSelectedCurrency)
+  const [sourceAmount] = useOnRampSourceAmount()
+  const { countryCode } = useLocale()
+
+  const {
+    data,
+    isLoading: isLoadingCryptoQuotes,
+    refetch,
+    isRefetching: isRefetchingCryptoQuotes,
+    error
+  } = useCreateCryptoQuote({
+    enabled:
+      countryCode !== undefined &&
+      selectedCurrency !== undefined &&
+      shouldCreateCryptoQuote &&
+      sourceAmount !== undefined &&
+      sourceAmount !== 0 &&
+      onRampToken?.currencyCode !== '' &&
+      onRampToken?.currencyCode !== undefined,
```

Comment:
this is a bit hard to read. I think this will be better
```
const hasValidCountry = countryCode !== undefined
const hasSelectedCurrency = selectedCurrency !== undefined
const hasSourceAmount = sourceAmount !== undefined && sourceAmount !== 0
...

const shouldFetchCryptoQuotes =
  shouldCreateCryptoQuote &&
  hasValidCountry &&
  hasSelectedCurrency && ...
```